src/ipa_pc/mod.rs

```diff
@@ -1814,19 +1814,69 @@ mod tests {
         use algebra::fields::tweedle::fr::Fr;
         use algebra::UniformRand;
 
-        let seed = b"";
-        let mut rng1 = FiatShamirChaChaRng::<Blake2s>::from_seed(seed);
-        let mut rng2 = FiatShamirChaChaRng::<Blake2s>::from_seed(seed);
+        // Empty test
+        {
+            let mut rng1 = FiatShamirChaChaRng::<Blake2s>::from_seed(FiatShamirChaChaRngSeed::default().finalize());
+            let mut rng2 = FiatShamirChaChaRng::<Blake2s>::default();
+
+            assert_eq!(rng1.get_state(), rng2.get_state());
+
+            let a = Fr::rand(&mut rng1);
+            let b = Fr::rand(&mut rng2);
+
+            assert_eq!(a, b);
+            assert_eq!(rng1.get_state(), rng2.get_state());
+
+            rng1.absorb(b"ABSORBABLE_ELEM");
+            rng2.absorb(b"ABSORBABLE_ELEM");
 
-        let a = Fr::rand(&mut rng1);
-        let b = Fr::rand(&mut rng2);
+            assert_eq!(rng1.get_state(), rng2.get_state());
 
-        assert_eq!(a, b);
+            let a: Fr = rng1.squeeze_128_bits_challenge();
+            let b: Fr = rng2.squeeze_128_bits_challenge();
+
+            assert_eq!(a, b);
+            assert_eq!(rng1.get_state(), rng2.get_state());
```

Comment:
May be UT for 'set_state' function should be added too?

Reply:
Done